hypervisor/arch/x86/guest/vm.c

```diff
@@ -407,8 +407,13 @@ int32_t create_vm(uint16_t vm_id, uint64_t pcpu_bitmap, struct acrn_vm_config *v
 
 	/* Allocate memory for virtual machine */
 	vm = &vm_array[vm_id];
-	/* the vm_state lock field need to remain unchanged in vm data structure */
-	(void)memset((void *)&vm->arch_vm, 0U, (sizeof(struct acrn_vm) - sizeof(spinlock_t)));
+	/*
+	 * the vm_state lock field need to remain unchanged in vm data structure
+	 * Since arch_vm struct is page aligned, size used for memeset should consider
+	 * subtracting 4K from the total size of acrn_vm.
+	 * Using offset_of macro to avoid hardcoding 4k here.
+	 */
+	(void)memset((void *)&vm->arch_vm, 0U, (sizeof(struct acrn_vm) - offsetof(struct acrn_vm, arch_vm)));
```

Comment:
I see. What we want to reset is from arch_vm to the end of vm structure.
Will  " (size_t *) (vm+1) - (size_t) &vm->arch_vm"  be easier to reader?

You decide.

Reply:
Oh, the current definition of struct acrn_vm may waste ~1 page of memory. We need a cleaner solution.